doc/use_cli_effectively.md

```diff
@@ -94,44 +95,132 @@ Most update commands in the CLI feature the three generic arguments: `--add`, `-
     az vm update -g my_rg -n my_vm --add storageProfile.dataDisks @~/my_disk.json
     ```
 
-## Quoting Issues ##
+## Generic resource commands - `az resource`
+
+There may be cases where a service you are interested in does not have CLI command coverage. You can use the `az resource create/show/list/delete/update/invoke-action` commands to work with these resources. A few suggestions here:
+1. If only `create/update` are involved, consider using `az group deployment create`. Leverage [Azure Quickstart Templates](https://github.com/Azure/azure-quickstart-templates) for working examples.
+2. Check out the Rest API reference for the request payload, URL and API version. As an example, check out the community's comments on [how to create AppInsights](https://github.com/Azure/azure-cli/issues/5543).
+
+## REST API command - `az rest`
+
+If neither generic update arguments nor `az resource` meets your needs, you can use `az rest` command to call the REST API. It automatically authenticates using the credential logged in and sets header `Content-Type: application/json`.
```

Comment:
consider, instead of "using the credential logged in" to "using the logged-in credential"

"and sets header" to "and sets the header to"

Reply:
```suggestion
If neither generic update arguments nor `az resource` meets your needs, you can use `az rest` command to call the REST API. It automatically authenticates using the logged-in credential and sets header `Content-Type: application/json`.
```

As for the header, it is setting one of the header entry, instead of setting all header entries.